server/api_archive.go

```diff
@@ -0,0 +1,126 @@
+package server
+
+// This is a low-level archiver which zips and sends GCA server persistent file
+// data using a POST command. This is intended to be used by an external service
+// to provide archival snapshots of raw data in the system. Only public data is
+// provided, with private keys zeroed before sending.
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/glowlabs-org/gca-backend/glow"
+)
+
+var (
+	ApiArchiveRateLimiter *glow.RateLimiter // Provide a rate limiter for this API.
+)
+
+func init() {
+	ApiArchiveRateLimiter = glow.NewRateLimiter(apiArchiveLimit, apiArchiveRate)
+}
+
+// ArchiveHandler provides the POST /archive api endpoint. It returns
+// uninterpreted file data intended to be used by an external service
+// to produce an archive.
+func (gcas *GCAServer) ArchiveHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
`/archive` should be a GET request because it's not changing any data

Reply:
General rule of thumb: if the caller knows they are changing server state, it's a POST request, everything else is a GET request. UPDATE and DELETE are not used because not all http libraries support those as first class use cases, but pretty much everything supports both GET and POST as first class use cases

Reply:
Makes sense, I think the reason for POST is that we are not returning exactly the data files after zeroing, however we are not actually changing he server state, so GET makes sense